src/components/VirtualizedAutocomplete/index.tsx

```diff
@@ -0,0 +1,277 @@
+import {
+	Autocomplete,
+	AutocompleteProps,
+	ChipTypeMap,
+	Box,
+	ListItemButton,
+	ListItemButtonProps,
+} from '@mui/material';
+import throttle from 'lodash/throttle';
+import * as React from 'react';
+import { VList } from 'virtua';
+
+const MUI_LIST_VERTICAL_PADDING = 8;
+
+interface ItemDataElement {
+	props: React.HTMLAttributes<HTMLElement>;
+	option: React.ReactNode;
+	index: number;
+}
+
+const ListboxComponent = ({
+	children,
+	isNextPageLoading,
+	...props
+}: React.HTMLAttributes<HTMLElement> & {
+	pagination: { loadNextPage?: () => Promise<void>; itemCount?: number };
+	isNextPageLoading: boolean;
+}) => {
+	const itemData = (children as ItemDataElement[]).slice();
+	const { itemCount, loadNextPage } = props.pagination;
+	const [optionHeightsTotal, setOptionHeightsTotal] = React.useState(
+		MUI_LIST_VERTICAL_PADDING * 2,
+	);
+
+	React.useEffect(() => {
+		if (!isNextPageLoading) {
+			setOptionHeightsTotal(MUI_LIST_VERTICAL_PADDING * 2);
+		}
+	}, [isNextPageLoading]);
+
+	React.useEffect(() => {
+		if (optionHeightsTotal < document.documentElement.clientHeight * 0.4) {
+			const options = document.getElementsByClassName('MuiAutocomplete-option');
+			let total = MUI_LIST_VERTICAL_PADDING * 2;
+			for (const option of options) {
+				total += option.clientHeight;
+			}
+			setOptionHeightsTotal(total);
+		}
+	}, [
+		document.documentElement.clientHeight,
+		optionHeightsTotal,
+		document.getElementsByClassName('MuiAutocomplete-option'),
+	]);
+
+	return (
+		<div
```

Comment:
lets replace this with a Box for consistency like https://github.com/balena-io-modules/ui-shared-components/blob/56c77fcfaae5e1128ef8ace73beed8ddf9b08e7b/src/components/CookiesBanner/index.tsx#L108